components/net/http_cache.rs

```diff
@@ -713,61 +813,32 @@ impl HttpCache {
                 constructed_response.referrer_policy = request.referrer_policy.clone();
                 constructed_response.raw_status = cached_resource.data.raw_status.clone();
                 constructed_response.url_list = cached_resource.data.url_list.clone();
-                cached_resource.data.expires = get_response_expiry(&constructed_response);
+
                 let mut stored_headers = cached_resource.data.metadata.headers.lock().unwrap();
                 stored_headers.extend(response.headers);
                 constructed_response.headers = stored_headers.clone();
-                return Some(constructed_response);
-            }
-        }
-        None
-    }
+                constructed_response
+            };
 
-    fn invalidate_for_url(&mut self, url: &ServoUrl) {
-        let entry_key = CacheKey::from_servo_url(url);
-        if let Some(cached_resources) = self.entries.get_mut(&entry_key) {
-            for cached_resource in cached_resources.iter_mut() {
-                cached_resource.data.expires = Duration::seconds(0i64);
-            }
-        }
-    }
+            cached_resource.write().data.expires = get_response_expiry(&constructed_response);
```

Comment:
note: assign result of `get_response_expiry` to a variable, then acquire the lock and mutate the resource.

Reply:
Done: just made the entire block `write`.